### PR TITLE
Update Windows Agent Installer to 1.7 and WinSW to 2.0.2

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This change integrates major WinSW fixes into Jenkins. The change is WiP, https://github.com/jenkinsci/windows-slave-installer-module/pull/6 to be integrated at least

### WinSW changes

The update includes many fixes and improvements, the full list is provided in the [WinSW changelog](https://github.com/kohsuke/winsw/blob/master/CHANGELOG.md). There are several issues referenced in Jenkins bugtracker:

* [JENKINS-22692](https://issues.jenkins-ci.org/browse/JENKINS-22692) - Connection reset issues when WinSW gets terminated due to the system shutdown
* [JENKINS-23487](https://issues.jenkins-ci.org/browse/JENKINS-23487)- Support of shared directories in WinSW
* [JENKINS-39231](https://issues.jenkins-ci.org/browse/JENKINS-39231) - Enable Runaway Process Killer by default
* [JENKINS-39237](https://issues.jenkins-ci.org/browse/JENKINS-39237) - Auto-upgrade of JNLP agent versions on the slaves

### Windows Agent Installer changes

* Adapt the default configurations to pick fixes above
* Slave => Agent renaming where possible

### Jenkins core changes

* Modify the configuration template, reference advanced options
* Enable Runaway Process Killer by default

@reviewbybees

### CODE REVIEW

File: core/src/main/resources/windows-service/jenkins-slave.xml [hunk 1]
```diff
@@ -23,15 +23,22 @@ THE SOFTWARE.
 -->
 
 <!--
-  Windows service definition for Jenkins slave
+  Windows service definition for Jenkins agent.
+  The actual version of this definition has been moved to https://github.com/jenkinsci/windows-slave-installer-module/blob/master/src/main/resources/org/jenkinsci/modules/windows_slave_installer/jenkins-slave.xml.
```
Comment (oleg-nenashev) [inline core/src/main/resources/windows-service/jenkins-slave.xml]: I do not see any usages, but this file can be downloaded as a resource.
So I'd prefer to keep it up to date

Comment (daniel-beck) [inline core/src/main/resources/windows-service/jenkins-slave.xml]: Left behind from https://github.com/jenkinsci/jenkins/blob/f79f53ffbdff78c8bb4fefb51c6acbb3f266001b/core/src/main/java/hudson/lifecycle/WindowsSlaveInstaller.java#L219 it seems -- probably time to remove it from core?

Comment (oleg-nenashev) [inline core/src/main/resources/windows-service/jenkins-slave.xml]: IMHO then we need to add the resource alias somehow

Comment (oleg-nenashev) [inline core/src/main/resources/windows-service/jenkins-slave.xml]: OK, I have found no legitimate way to use this file excepting the access to the master's classloader from plugins (why would they do it). No occurrence in OSS repos as well.

Will remove the file

File: war/pom.xml [hunk 2]
```diff
@@ -114,7 +114,7 @@ THE SOFTWARE.
     <dependency>
       <groupId>org.jenkins-ci.modules</groupId>
       <artifactId>windows-slave-installer</artifactId>
-      <version>1.6</version>
+      <version>1.7-SNAPSHOT</version>
```
Comment (daniel-beck) [inline war/pom.xml]: Could you upload a snapshot and reference it by timestamp?

**Comment by oleg-nenashev** [comment]
Created PR with some upgrade docs: https://github.com/jenkinsci/windows-slave-installer-module/pull/9

**Comment by oleg-nenashev** [comment]
@jenkinsci/code-reviewers since I was unable to get feedback from @reviewbybees in time :(

**Comment by oleg-nenashev** [comment]
@reviewbybees ping. 1-week review timeout is today

File: core/pom.xml [hunk 3]
```diff
@@ -755,7 +755,7 @@ THE SOFTWARE.
                 <artifactItem>
                   <groupId>com.sun.winsw</groupId>
                   <artifactId>winsw</artifactId>
-                  <version>1.16</version>
+                  <version>2.0.2</version>
```
Comment (jglick) [inline core/pom.xml]: [Full diff](https://github.com/kohsuke/winsw/compare/1.16...winsw-2.0.2)

File: war/pom.xml [hunk 4]
```diff
@@ -114,7 +114,7 @@ THE SOFTWARE.
     <dependency>
       <groupId>org.jenkins-ci.modules</groupId>
       <artifactId>windows-slave-installer</artifactId>
-      <version>1.6</version>
+      <version>1.7</version>
```
Comment (jglick) [inline war/pom.xml]: [Full diff](https://github.com/jenkinsci/windows-slave-installer-module/compare/windows-slave-installer-1.6...windows-slave-installer-1.7)

**Review by jglick — APPROVED** [review]
I presume you have done manual testing on various platforms. Given the magnitude of the changes, meaningful code review seems hopeless.

**Comment by jglick** [comment]
```
21:08:37 [Windows] [ERROR] npm ERR! code ECONNRESET
```

is likely CI flakiness.

**Comment by oleg-nenashev** [comment]
@jglick 

> I presume you have done manual testing on various platforms. Given the magnitude of the changes, meaningful code review seems hopeless.

Yes, tested changes on 3 platforms (Win 10, Win Server 2012, Win 7).
I would also like to test the stuff on WinXP, but I have no such machine. In the case of WinSW there was one bug report to this platform, which has been fixed in 2.0.2

@reviewbybees done

**Comment by oleg-nenashev** [comment]
Got +1 from @daniel-beck in IRC

**Pull request merged**